### PR TITLE
Docker setup for development

### PR DESCRIPTION
**Pull request by Yutsuten**
As I said at #10019 these are the new docker files for development.
Still a work in progress, seems that it still have some glitches on the client. I'll search that later.

**How to use:**
Start with `docker-compose -f docker-compose.dev.yml up -d`
See the logs with `docker-compose logs -f --tail 20`
Stop with `docker-compose down`

While testing, the server should update as you change the code locally.

----
UUID: 203f208a-e0f0-4442-bf16-d4f19e553e8a

### CODE REVIEW

**Comment by Yutsuten** [comment]
Looks good now, ready for merge.

File: Dockerfile-Dev [hunk 1]
```diff
@@ -1,18 +1,5 @@
-FROM node:10
-
-# Install global packages
-RUN npm install -g gulp-cli mocha
-
-# Clone Habitica repo and install dependencies
-RUN mkdir -p /usr/src/habitrpg
-WORKDIR /usr/src/habitrpg
-RUN git clone https://github.com/HabitRPG/habitica.git /usr/src/habitrpg
-RUN cp config.json.example config.json
```
Comment (paglias) [inline Dockerfile-Dev:10]: Is using custom config values in config.json still supported?

Comment (Yutsuten) [reply Dockerfile-Dev:10]: Sure. All the source code, including the `config.json`, are being shared from the local source code to the containers.
The only thing that is not synced in real time is the `node_modules` folder. If there are new packages, we should rebuild the docker image.

**Comment by paglias** [comment]
I've assigned this to @SabreCat who's more familiar with our Docker setup but it looks good to me!

**Comment by SabreCat** [comment]
Works a treat! Thanks for this, @Yutsuten -- crediting this toward your next contrib tier!

**Comment by GrimmiMeloni** [comment]
@Yutsuten I fiddled around 2 weeks ago (i.e. prior to your change) with getting the docker based setup working, but failed. I checked and see that the docker description in the [wiki](https://habitica.fandom.com/wiki/Setting_up_Habitica_Locally#Docker) has not changed, yet. Is there any new documentation somewhere else, that accompanies these changes? 

I am asking, because looking at the dev Dockerfile changes you did, my interpretation would be that we are now expected to put the code that goes into the container into a subdirectory called `code`?
Do we still need to mount the local code on the docker host into the container like described on the wiki?

**Comment by Yutsuten** [comment]
@GrimmiMeloni the wiki needs to be updated, probably all the information there is wrong now.

The basic usage I wrote in the first message of this PR, nothing else should be necessary.
Syncing the files between local and the container is done automatically by docker and docker-compose.

**Comment by rsamborski** [comment]
@Yutsuten thanks for updating docker-compose environment. Unfortunetely I am trying to run it with current develop branch, but it always fails the server part. The logs are not very useful (I turned on verbose mode):

```
server_1  | [20:14:07] Using gulpfile /code/gulpfile.js
server_1  | [20:14:07] Starting 'nodemon'...
server_1  | [20:14:07] Finished 'nodemon' after 94 ms
server_1  | [20:14:07] [nodemon] 1.19.1
server_1  | [20:14:07] [nodemon] to restart at any time, enter `rs`
server_1  | [20:14:07] [nodemon] watching: *.*
server_1  | [20:14:07] [nodemon] starting `node ./website/server/index.js`
server_1  | npm verb lifecycle habitica@4.109.1~start: unsafe-perm in lifecycle true
server_1  | npm verb lifecycle habitica@4.109.1~start: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/code/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
server_1  | npm verb lifecycle habitica@4.109.1~start: CWD: /code
server_1  | npm info lifecycle habitica@4.109.1~start: Failed to exec start script
server_1  | npm verb stack Error: habitica@4.109.1 start: `gulp nodemon`
server_1  | npm verb stack Exit status 1
server_1  | npm verb stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:326:16)
server_1  | npm verb stack     at EventEmitter.emit (events.js:209:13)
server_1  | npm verb stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
server_1  | npm verb stack     at ChildProcess.emit (events.js:209:13)
server_1  | npm verb stack     at maybeClose (internal/child_process.js:1021:16)
server_1  | npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
server_1  | npm verb pkgid habitica@4.109.1
server_1  | npm verb cwd /code
server_1  | npm verb Linux 5.0.0-25-generic
server_1  | npm verb argv "/usr/local/bin/node" "/usr/local/bin/npm" "start" "--verbose"
server_1  | npm verb node v12.9.0
server_1  | npm verb npm  v6.10.2
server_1  | npm ERR! code ELIFECYCLE
server_1  | npm ERR! errno 1
server_1  | npm ERR! habitica@4.109.1 start: `gulp nodemon`
server_1  | npm ERR! Exit status 1
server_1  | npm ERR!
server_1  | npm ERR! Failed at the habitica@4.109.1 start script.
server_1  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
server_1  | npm verb exit [ 1, true ]
server_1  | npm timing npm Completed in 22512ms
server_1  |
server_1  | npm ERR! A complete log of this run can be found in:
server_1  | npm ERR!     /root/.npm/_logs/2019-08-31T20_14_21_198Z-debug.log
habitica_server_1 exited with code 1
```

Docker versions:
```
Docker version 18.06.1-ce, build e68fc7a
docker-compose version 1.22.0, build unknown
```

Any ideas on what can be the issue? I tried on two different Ubuntu machines (one a dedicated server and second a VirtualBox machine).

**Pull request merged**